app/controllers/admin/tutorials_controller.rb

```diff
@@ -38,11 +39,35 @@ def tutorial_params
     params.require(:tutorial).permit(:tag_list)
   end
 
-  def results
-    YoutubePlaylistResults.new(params[:playlist_id])
+  def valid_thumbnail?(thumbnail)
+    thumbnail = thumbnail.downcase
+    valid_thumbnail1 = 'http://img.youtube.com'
+    valid_thumbnail2 = 'https://img.youtube.com'
+    valid_thumbnail3 = 'img.youtube.com'
+    scenario1 = thumbnail[0..21] == valid_thumbnail1[0..21]
+    scenario2 = thumbnail[0..22] == valid_thumbnail2[0..22]
+    scenario3 = thumbnail[0..14] == valid_thumbnail3[0..14]
+
+    scenario1 || scenario2 || scenario3
   end
 
+  # def results
+  #   YoutubePlaylistResults.new(params[:playlist_id])
+  # end
+
   def new_tutorial_params
-    results.parameters
+    params.require(:tutorial).permit(:title, :description, :thumbnail)
+  end
+
+  def user_params
+    params.require(:user).permit(:email, :first_name, :last_name, :password)
+  end
+
+  def missing_params
+    missing_params = []
+    params[:tutorial].each do |key, value|
+      missing_params << key if value == ''
+    end
+    missing_params.join(', ')
```

Comment:
just creating a string of params that are ""

Reply:
this is cool